### ISSUE TITLE
Add a contributing guide

### ISSUE THREAD

**Issue by pragmaticivan** [elixirschool/extracurricular#39]
As a new possible contributer, I got a bit lost about what are the required dependencies or why there's a lack of fake data for seed and local test.

Maybe we can explain a bit how to get the app up and running and see some fake that so the user can test locally.

**Comment by doomspork** [comment]
Howdy @pragmaticivan!  We're still working on our process for this so feedback is welcome.  

TL;DR:  We've intentionally left off things so that we can have conversation around them and work directly with people, who haven't done open source and want to, on their implementation.  At the end of the day we want to build a useful product but also get people involved with open source who otherwise wouldn't.

Long winded response:

This is a very new project that we've decided to approach a little differently.  In order to get people involved who might otherwise not, @ybur-yug and myself are using our time to work hands-on with anyone interested in contributing.  This way we work directly with them to understand the scope, the project goals, work through a solution, and leave behind comments and feedback that would be useful to someone else following along.

One concern I have, that @ybur-yug may share, is that if we invest too much time upfront writing documentation and features some potential negatives can occur for which there are fewer positives:

+ Experienced people swoop in and pick up all the low hanging fruit.  We see this as a negative because it eliminates opportunities for less experienced people to be involved.  See my comments on #35; I would prefer experienced people avoid even logging bugs before they're an issue because the process of finding a bug is part of the learning experience.
+ Between @ybur-yug and I we only have a very limited about my free time.  We've decided that time is better spent working directly with contributors than it is on writing documentation and planning out future features.  Without forward progress we run the risk of people getting bored and the project stalemating.  This is a big negative because we lose out on a useful tool and people who've already committed their time will have done so for naught, the result of which is that they may not contribute to open source going forward.
+ Stories will get started and contributors won't be available to incorporate feedback in a timely manner.  This could result in the project being blocked indefinitely or us having to finish people's work for them.  This directly impacts our ability to launch the project which is why we've elected to work on the MVP feature set in the way we have. 
+ The other concern is that if people read a story, think they understand all the pieces, and start on their work in a vacuum they may be duplicating work done by someone else or they may not even fully understand the story.  We're trying to force people to get involved and have discussions.

Before we can spend too much time on anything else it's absolutely critical we actually get the MVP out and for that @ybur-yug and I are working with people who can commit the necessary time.  If you've got time let me know and we can definitely figure out some things for you to bang out 😀 

All of these things are in the pipeline and soon to come.  Priority number one is getting the very basic set of features out-the-door so people can see what they're contributing too.  Having the product live and being used will also inform the next set of features.

I'm going to spend some time this week expanding on the README so it's clearer to folks such as yourself and @jeffweiss what our goals are and why we're doing things the way we are (such as refraining from logging bugs or features beyond the MVP).

Let me know if you've got any questions/comments/concerns @pragmaticivan!

**Comment by pragmaticivan** [comment]
That sound really good to me! 👍 

But I still think that telling the newcomers this new philosophy in a `CONTRIBUTING.md` and at least telling the user that they need postgres, redis, etc... installed, and at least have a bit of `fake data` to bootstrap and see something working would optimize you guys time.

Just wondering, let's move on with that you guy guys suggested, I like it 🍻

**Comment by doomspork** [comment]
@pragmaticivan I will _definitely_ work on those after we get the MVP out this.  Getting this launched is a milestone that sort of makes the whole thing "real".  We want to hit that goal then take a step back, look at our process, discuss it, and improve it 😀 

I should add that an org-wide contributing guide is one of the first blog posts I want to write. 

I hear regularly from folks that contributors to Elixir School are so welcoming and ego-less that it makes people want to come back and contribute more, we should ensure that continues to be the case.

Thanks again for all of your help and continued interest @pragmaticivan, you da :bomb:

**Comment by doomspork** [comment]
@pragmaticivan now that we've merged in the guide to Campus and updated the README here, what else would you like to see added?  At the very least we should add a link from this README to Campus'

**Comment by pragmaticivan** [comment]
The current `README.md` looks good to me. Thanks! o/

**Comment by doomspork** [comment]
Wonderful!  Thanks again for your interest @pragmaticivan.  I'm going to get this deployed to Digital Ocean this week 😍
